group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupCoordinatorService.java

```diff
@@ -523,9 +526,38 @@ public CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection
             return FutureUtils.failedFuture(Errors.COORDINATOR_NOT_AVAILABLE.exception());
         }
 
-        return FutureUtils.failedFuture(Errors.UNSUPPORTED_VERSION.exception(
-            "This API is not implemented yet."
-        ));
+        final Map<Integer, List<String>> groupsByPartition = new HashMap<>();
+        for (String groupId : groupIds) {
+            final int partition = partitionFor(groupId);
+            final List<String> groupList = groupsByPartition.getOrDefault(partition, new ArrayList<>());
+            groupList.add(groupId);
+            groupsByPartition.put(partition, groupList);
+        }
+
+        final List<CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection>> futures = new ArrayList<>();
+        for (Map.Entry<Integer, List<String>> entry : groupsByPartition.entrySet()) {
+            int partition = entry.getKey();
+            List<String> groupList = entry.getValue();
+            CompletableFuture<DeleteGroupsResponseData.DeletableGroupResultCollection> future =
+                    runtime.scheduleWriteOperation("delete-group",
+                            new TopicPartition(Topic.GROUP_METADATA_TOPIC_NAME, partition),
+                            coordinator -> coordinator.deleteGroups(context, groupList));
+            futures.add(future);
+        }
+
+        final CompletableFuture<Void> allFutures = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
```

Comment:
Let's assume that one of the write operation fails with `COORDINATOR_LOAD_IN_PROGRESS`, this would result in failing `allFutures` even though some write operations may have been successful. It seems to me that we should handle exceptions for each write operation future before we combine them, no?

Reply:
Yeah, it makes sense. I'm not sure what to return if there's an exception in the write operation, since we can't set an error code for a `DeletableGroupResultCollection`.

Reply:
`DeletableGroupResultCollection` contains `DeletableGroupResult` which has an error code. Therefore I think that we should create a `DeletableGroupResult` per group id in the `groupList` when there is an exception.